src/crypto/index.js

```diff
@@ -319,6 +321,44 @@ Crypto.prototype.init = async function(opts) {
     this._checkAndStartKeyBackup();
 };
 
+/**
+ * Whether to trust a others users signatures of their devices.
+ * If false, devices will only be considered 'verified' if we have
+ * verified that device individually (effectively disabling cross-signing).
+ *
+ * Default: true
+ *
+ * @return {bool} True if trusting cross-signed devices
+ */
+Crypto.prototype.getCryptoTrustCrossSignedDevices = function() {
+    return this._trustCrossSignedDevices;
+};
+
+/**
+ * See getCryptoTrustCrossSignedDevices
+
+ * This may be set before initCrypto() is called to ensure no races occur.
+ *
+ * @param {bool} val True to trust cross-signed devices
+ */
+Crypto.prototype.setCryptoTrustCrossSignedDevices = function(val) {
+    this._trustCrossSignedDevices = val;
+
+    for (const userId of this._deviceList.getKnownUserIds()) {
+        const devices = this._deviceList.getRawStoredDevicesForUser(userId);
+        for (const deviceId of Object.keys(devices)) {
+            const deviceTrust = this.checkDeviceTrust(userId, deviceId);
+            if (
+                !deviceTrust.isLocallyVerified() &&
```

Comment:
How about the device is trusted locally but not by cross-signing? Should we also emit then? Or perhaps we fallback to local trust elsewhere when cross-signing is enabled?

Reply:
Yeah, this only changes the return value of `isVerified()` which is always true if `isLocallyVerified()` is true. Have commented.